bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
The implementation of this method will never return a null element. I think the annotation is an overweight of the method. IMHO it is better to update java doc to said the method returns not null value instead of adding  a runtime check on every method call.